src/transformers/models/byt5/tokenization_byt5.py

```diff
@@ -196,71 +206,45 @@ def build_inputs_with_special_tokens(
 
     def _tokenize(self, text: str) -> List[str]:
         """Take as input a string and return a list of strings (tokens) for words/sub-words"""
-
-        def _sub_tokenize(sub_text):
-            character_list = list(sub_text)
-            utf_tokens_lists = [list(char.encode("utf-8")) for char in character_list]
-            sub_tokens = [chr(utf_token) for utf_tokens in utf_tokens_lists for utf_token in utf_tokens]
-            return sub_tokens
-
-        # split on special characters
-        pattern = f"({'|'.join(self.special_tokens_encoder.keys())})"
-        sub_texts = list(filter(None, re.split(pattern, text)))
-        tokens = []
-        for sub_text in sub_texts:
-            if sub_text in self.special_tokens_encoder.keys():
-                tokens += [sub_text]
-            else:
-                tokens += _sub_tokenize(sub_text)
-
+        tokens = list(text)
```

Comment:
Think we need to split on special characters here no? Otherwise `</s>` gets incorrectly tokenized...Also what about new tokens?

Reply:
`tokenizer.tokenize` (within tokenization utils) already takes care of all special_tokens (though `self.unique_no_split_tokens`)